File: Wipro_Capstone_project/BACKEND/product-ms/product-ms/src/main/java/com/backend/product_ms/entity/Product.java
package com.backend.product_ms.entity;

import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Entity
@Data
public class Product {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int productId;
    
    @NotBlank
    private String productName;
    
    
    @NotBlank
    private String productDesc;
    
    @NotBlank
    private String productCat;
    
    @NotBlank
    private String make;
    
    @NotNull
    private Integer productAvailableQty; // use Integer instead of int for validation
    
    @NotNull
    private Double productPrice;
    
    @NotBlank
    private String productImgUrl;
    
    private LocalDate dateOfManufacture;
}

File: Wipro_Capstone_project/FRONTEND/CapstoneProjectAngular/src/app/interface/product.ts
// src/app/interface/product.ts

export interface Product {
  productId?: number;            // optional for new products
  productName: string;
  productDesc: string;
  productCat: string;
  make: string;
  productAvailableQty: number;
  productPrice: number;
  productImgUrl: string;
  dateOfManufacture?: string;    // use string for LocalDate from backend
}

// src/app/interface/product.ts
export interface ProductInOrder extends Product {
  quantityOrdered: number;
}

